toolkit/imageconfigs/marketplace-gen2.json

```diff
@@ -51,7 +52,8 @@
                 },
                 {
                     "ID": "rootfs",
-                    "MountPoint": "/"
+                    "MountPoint": "/",
+                    "MountIdentifier": "partlabel"
```

Comment:
One thing to double check here is if cloud-init growpart still functions with this change. In the past we've seen growpart fail because it couldn't identify the root partition when specified with PARTLABEL.

Please create an Azure VM with this new image and an explicit "--os-disk-size-gb 100" and make sure that the rootfs has been automatically expanded to fill all of that disk space.

Reply:
I tested with "--os-disk-size-gb 100" and the rootfs was indeed automatically expanded to fill all of the disk space.

But as Daniel suggested below, I figured out a way to make kata's grub.cfg work with only partition name and no partlabel. So I just reverted the partlabel change,